darwin/exporter/formats/yolo_segmented.py

```diff
@@ -0,0 +1,236 @@
+    """
+    Exports YoloV8 format as segments
+
+    Parameters
+    ----------
+    annotation_files : Iterable[AnnotationFile]
+        The ``AnnotationFile``\\s to be exported.
+    output_dir : Path
+        The folder where the new pascalvoc files will be.
+
+    Returns
+    -------
+    None
+    """
+    annotation_files = list(annotation_files)
+
+    class_index: ClassIndex = build_class_index(
+        # fmt: off
+        annotation_files, ["bounding_box", "polygon"]
+    )  # fmt: on
+
+    for annotation_file in annotation_files:
+        export_file(annotation_file, class_index, output_dir, _build_text)
+
+    save_class_index(class_index, output_dir)
+
+
+def normalise(value: float, height_or_width: int) -> float:
+    """
+    Normalises the value to a proportion of the image size
+
+    Parameters
+    ----------
+    value : float
+        The value to be normalised.
+    height_or_width : Union[float, int]
+        The height or width of the image.
+
+    Returns
+    -------
+    float
+        The normalised value.
+    """
+    return value / height_or_width
+
+
+class YoloSegmentedAnnotationType(Enum):
+    """
+    The YoloV8 annotation types
+    """
+
+    UNKNOWN = auto()
+    BOUNDING_BOX = auto()
+    POLYGON = auto()
+
+
+def _build_text(annotation_file: AnnotationFile, class_index: ClassIndex) -> str:
+    """
+    Builds the YoloV8 format as segments
+
+    Parameters
+    ----------
+    annotation_file : AnnotationFile
+        The ``AnnotationFile`` to be exported.
+    class_index : ClassIndex
+        The class index.
+
+    Returns
+    -------
+    str
+        The YoloV8 format as segments
+    """
+    yolo_lines: List[str] = []
+
+    im_w = annotation_file.image_width
+    im_h = annotation_file.image_height
+
+    if not im_w or not im_h:
+        raise ValueError(
+            "Annotation file has no image width or height. "
+            "YoloV8 Segments are encoded as a proportion of height and width. "
+            "This file cannot be YoloV8 encoded without image dimensions."
+        )
+
+    for annotation_index, annotation in enumerate(annotation_file.annotations):
+        if isinstance(annotation, VideoAnnotation):
+            logger.warn(
+                f"Skipped annotation at index {annotation_index} because video annotations don't contain the needed data."
+            )
+            continue
+
+        if "x" in annotation.data and "y" in annotation.data and "w" in annotation.data and "h" in annotation.data:
+            annotation_type = YoloSegmentedAnnotationType.BOUNDING_BOX
+        elif "points" in annotation.data:
+            if isinstance(annotation.data["points"][0], list):
+                logger.warn(f"Skipped annotation at index {annotation_index} because it's a complex polygon'")
+                continue
+
+            annotation_type = YoloSegmentedAnnotationType.POLYGON
+        else:
+            annotation_type = YoloSegmentedAnnotationType.UNKNOWN
```

Comment:
might be useful to extract to a helper func, improves the readability then we can do something like:

```python
annotation_type = get_annotation_type(annotation.data)
```

Reply:
Yeah, agreed.  In fact I only didn't because of the time frame.  The original plan had a helper func for this, and separate predicate functions for handling bounding boxes and polygons